### ISSUE TITLE
Multiple constraints dependencies does not work and poetry faile with a RecursionError

### ISSUE THREAD

**Issue by gruel** [python-poetry/poetry#2410]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Linux Archlinux
- **Poetry version**: 1.0.5
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/gruel/6a136d198f4f8dd6ab77904cc307b3fb

## Issue
Poetry is going in an RecursionError when using markers to specify a different version for a specific system plateform. It seems that the markers is not used and Poetry is entering in a loop. 

In addition to the problem that specify a system_plateform seems to be useless, it seems also that: ^5.14 and 5.14.0 should find a common solution but it is not the case.

[poetry install -vvv output](https://gist.github.com/gruel/09a5de16c2d930ea929dc50868065d49)

**Comment by RCJacH** [comment]
I'm experiencing similar result.
- OS version and name: Windows 10
- Poetry version: 1.0.5


Below is a minimal project that triggers the problem:

```
poetry new test
cd test
poetry remove --dev pytest #This is used to decrease the number of recursing lines
poetry add moviepy
```

Then it loops with the following lines:
```
   0: Retrying dependency resolution for Python (>=3.8,<4.0).
   1: fact: test is 0.1.0
   1: derived: test
   1: fact: test depends on moviepy (^1.0.3)
   1: selecting test (0.1.0)
   1: derived: moviepy (^1.0.3)
   0: Duplicate dependencies for numpy
   0: Different requirements found for numpy (>=1.17.3) and numpy (*).
   1: Version solving took 0.055 seconds.
   1: Tried 1 solutions.
```

In my case, the issue is with `0: Different requirements found for numpy (>=1.17.3) and numpy (*).`

**Comment by g4b1nagy** [comment]
This has been fixed in the `1.1.X` branch of `Poetry`. So every version starting from `1.1.0` (Oct 1, 2020) should work just fine.
Note that:
> lock files generated with this release are not compatible with previous releases of Poetry.

**Comment by finswimmer** [comment]
Thanks for your response @g4b1nagy . So I will close this. If anyone disagree please leave a comment.
